enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+
+If a resource has multiple `capability.openshift.io` annotations, then the resource will be filtered any
+of the annotations on the resource are matched by a configured filter.  This allows a component to define
+a `ServiceMonitor` which is part of both the `Monitoring` capability and the `Foo` capability.  If either
+of those capabilities are excluded, the `ServiceMonitor` should not be created.  (If monitoring is excluded
+then no ServiceMonitors should be created, if only Foo is excluded, then only its ServiceMonitors should
+be excluded).
+
+
+If the install-config specified `none` as the excluded component, the CVO list will be empty.  If the
+install-config specified `default`, the CVO list will be whatever the installer chooses to disable for
+the install.  The CVO does not need to be aware of those special keywords.  Specifying any additional
+values along with `none` is invalid.
+
+4) Admin can remove an item from the filtered annotations list, but they cannot add an item to it.  If an
+item is removed, the CVO will apply the previously filtered(skipped) resources to the cluster on the next reconciliation.
+Adding an item to the filtered list is not supported because it requires the component be removed from the
+running cluster which has more significant implications for how all traces of the component are removed.  The
+CVO should reject updates that add new entries to the filter list.
+
+The currently configured filter list for the CVO should be recorded in telemeter so we can understand
+the configuration of a given cluster.
+
+If we want components to be disabled by default (either existing components or new ones added in the future),
+we can add their component names to a list of default-disabled components that the installer populates
+the install-config with.  Users can then edit that disabled list in the install-config to remove those
+components if they want them enabled, and add other components they'd like to disable.
+
+In the future, we might allow specific APIs to be disabled, such as the build api.  This could be done by
+defining additional capability keywords that can be put into the install-config field being defined here,
+which would drive the creation of cluster config that the apiserver+controllers used to disable that particular
+api.
+
+
+### Risks and Mitigations
+
+The primary risk is that teams understand how to use these new annotations and apply them
+correctly to the full set of resources that make up their component.  Inconsistent or
+partial labeling will result in inconsistent or partially deployed resources for a component.
+
+Another risk is that this introduces more deployment configurations which might
+have unforeseen consequences (e.g. not installing the imageregistry causes some
+other component that assumes there is always an imageregistry or assumes the
+presence of some CRD api that is installed with the imageregistry to break).
```

Comment:
Can you think of a scenario where we test with X removed and the cluster is ok, and when we test with Y removed the cluster is ok, but when we test with X+Y removed, the cluster will not be ok?

(also we can't blindly run our existing test suites on clusters with components removed, since not all tests will be applicable).

So yes, we need to be careful about what components we allow to be disabled, and do some amount of reasoning/testing around clusters that have that component disabled.  But I do not think we need to test every combination of disabled/enabled components.  The primary use of this feature will be to remove leaf components(ones that are not depended upon by another component)

Reply:
Sounds good to me. I think as long as we install critical components in any install configurations and the default installation experience works we should be fine.